config/yaml_config.go

```diff
@@ -75,19 +77,33 @@ func (a *AgentConfig) loadNetworkYamlConfig(path string) error {
 	// Pull additional parameters from the global config file.
 	a.LogLevel = config.Datadog.GetString("log_level")
 	a.StatsdPort = config.Datadog.GetInt("dogstatsd_port")
+	if config.Datadog.IsSet("bind_host") {
+		a.StatsdHost = config.Datadog.GetString("bind_host")
+	}
 
 	return nil
 }
 
 // Process-specific configuration
-func (a *AgentConfig) loadProcessYamlConfig(path string) error {
-	config.Datadog.AddConfigPath(path)
-	if strings.HasSuffix(path, ".yaml") { // If they set a config file directly, let's try to honor that
-		config.Datadog.SetConfigFile(path)
-	}
+func (a *AgentConfig) loadProcessConfig(iniPath, yamlPath string) error {
```

Comment:
Since this takes care of loading the ini config aswell should we rename the file ?

Reply:
I was going to merge this all into `config.go` after the environment piece is done to reduce noise :)